### PR TITLE
SALTO-5789: remove workflowV2 warnings when workflow version is workflowV2

### PR DESCRIPTION
**Pull request by shayc331**
_remove workflowV2 warnings when workflow version is workflowV2_

---

_Additional context for reviewer_

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

**Review by IdoZakk — COMMENTED** [review]
🦞 
I think it is a patchy solution. I know you have spent way too much time on those workflows, but let's try to make this right.

Currently you use `isWorkflowV2Transition` for two things- decide if it is new or old workflow and verify the structure of the transition, but only for V2....
 I can think of several other solutions. 
 -If we want to verify the structure we can do it when we verify the general workflow structure
 - We can check V2 structure, if it fails check V1, and if both fail issue a warning
 - We can split to two functions. One to check the version (which you take from outside), the second one to check structure in case it is V2

File: packages/jira-adapter/src/filters/workflow/transition_structure.ts [hunk 1]
```diff
@@ -17,13 +17,27 @@
...
+import { WorkflowStatusAndPort, WorkflowTransitionV2 } from '../workflowV2/types'
 
 const { makeArray } = collections.array
 
+const TRANSITION_FROM_V2_SCHEME = Joi.array().items(
+  Joi.object({
+    statusReference: Joi.alternatives(Joi.object(), Joi.string()).required(),
+    port: Joi.number(),
+  }),
+)
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/workflow/transition_structure.ts:33]: Isn't the array required? If so better to add it here. Also making sure the object cannot contain more fields

File: packages/jira-adapter/src/filters/workflow/transition_structure.ts [hunk 2]
```diff
@@ -17,13 +17,27 @@
...
+
+// we already validate the workflow structure in workflow_filter, so we just want to differ between the two versions
+const isTransitionFromV2 = (value: unknown): value is WorkflowStatusAndPort[] => {
+  const { error } = TRANSITION_FROM_V2_SCHEME.validate(value)
+  return error === undefined
+}
+
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/workflow/transition_structure.ts:40]: Check to use schemeGuard without the error message

**Review by IdoZakk — APPROVED** [review]
Great, look at the comments, and maybe add a test to make sure we cover both cases (V1, V2) properly

File: packages/jira-adapter/src/filters/workflow/transition_structure.ts (hunk 1 above)
Comment (shayc331) [reply packages/jira-adapter/src/filters/workflow/transition_structure.ts:33]: 1. the array is not required, for example global transition does not have `from`
2. yes

**Pull request merged**